aspnetcore/fundamentals/primitives/change-tokens.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Detect changes with Change Tokens in ASP.NET Core
+author: guardrex
+description: Learn how to use Change Tokens to track changes.
+ms.author: riande
+manager: wpickett
+ms.date: 11/10/2017
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/primitives/change-tokens
+---
+# Detect changes with Change Tokens in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+A Change Token is a general-purpose, low-level building block used to track changes.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/primitives/change-tokens/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## IChangeToken interface
+
+The public interface [IChangeToken](/dotnet/api/microsoft.extensions.primitives.ichangetoken) propagates notifications that a change has occurred. `IChangeToken` resides in the [Microsoft.Extensions.Primitives](/dotnet/api/microsoft.extensions.primitives) namespace. For apps that don't use the [Microsoft.AspNetCore.All](https://www.nuget.org/packages/Microsoft.AspNetCore.All/) metapackage, reference the [Microsoft.Extensions.Primitives](https://www.nuget.org/packages/Microsoft.Extensions.Primitives/) NuGet package in the project file.
+
+The interface has two properties:
+
+* [ActiveChangedCallbacks](/dotnet/api/microsoft.extensions.primitives.ichangetoken.activechangecallbacks) indicate if the token proactively raises callbacks. Callbacks are still guaranteed to fire, eventually. 
```

Comment:
`Callbacks are still guaranteed to fire, eventually.`? No they are not. If this is false then a callback will never be called, you need to poll `HasChanged` for changes. It's also possible a token will never be cancelled.

Reply:
@Tratcher I'll address this for the topic but note that it's in the comments:

https://github.com/aspnet/Common/blob/c9f4312474fa685f289f82d69288cd29b8a4f01a/src/Microsoft.Extensions.Primitives/IChangeToken.cs#L19